### DISCUSSION TITLE
Property based convergence criteria

### DISCUSSION THREAD

**Discussion by mjohnson541** [ReactionMechanismGenerator/RMG-Py#2526] [Ideas]
I was looking back at issue #2521 and thought to myself, why don't we decide if an RMG model is converged based on the convergence of properties the user chooses automatically? I think part of the the old reason is that RMG only had a very limited set of reactors (couldn't do this for ignition delay for example), which isn't as much of an issue anymore with the RMS reactors. 

The design isn't trivial...you definitely don't want to stop the first iteration a property doesn't change, but as this is in practice how we decide if RMG runs are converged it should be highly practical to just have RMG do this automatically.

**Comment by JacksonBurns** [comment]
What's an example of a property that someone might want to use as a convergence criteria?

**Comment by mjohnson541** [comment]
First coming to mind: Ignition delay, concentration of a species (at a given time or range of times), thermodynamic variables (T,P etc. at a given time or range of times) for catalysis probably the TOF.
